python/paddle/tensor/linalg.py

```diff
@@ -430,7 +430,7 @@ def inf_norm(input,
         reduce_all = True if axis == None or axis == [] or asvector == True else False
         axis = axis if axis != None and axis != [] else [0]
 
-        reduce_type = 'reduce_max' if porder == np.float(
+        reduce_type = 'reduce_max' if porder == np.float64(
```

Comment:
This should also to be np.float32 as porder is defined as "porder (float, optional): None for porder=2.0." in code here:

https://github.com/PaddlePaddle/Paddle/blob/develop/python/paddle/tensor/linalg.py#L378

@ZHUI to double confirm, thanks!

Reply:
https://github.com/PaddlePaddle/Paddle/blob/develop/paddle/phi/kernels/p_norm_kernel.h#L24 

Yes, it should be `np.float32 ` since the p_norm kernel use `float` type.